rpc/src/rpc/native/block.rs

```diff
@@ -0,0 +1,34 @@
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use anyhow::{anyhow, Result};
+use ethers::providers::{Http, Provider, RetryClient};
+use ethers::types::H256;
+use tokio::sync::Mutex;
```

Comment:
> Contrary to popular belief, it is ok and often preferred to use the ordinary [Mutex](https://doc.rust-lang.org/nightly/std/sync/mutex/struct.Mutex.html) from the standard library in asynchronous code.

_[`tokio::sync::Mutex` docs](https://docs.rs/tokio/latest/tokio/sync/struct.Mutex.html#which-kind-of-mutex-should-you-use)_

They generally don't work well with cancellation, defy user expectations, maybe check out [this fragment](https://youtu.be/1zOd52_tUWg?si=kis16mw-8ReoEG5P&t=2003), or the above docs.

I think the most appropriate changes is to refactor the code so that no shared state is needed - can this be a map reduce instead?

Reply:
updated to map reduce